flocker/node/agents/blockdevice.py

```diff
@@ -574,6 +575,29 @@ def run(self, deployer):
         return run_state_change(sequentially(changes=changes), deployer)
 
 
+def _create_mount_directory(mountpoint):
+    """
+    Create the directory where a device will be mounted.
+
+    The directory's parent's permissions will be set to only allow access
+    by owner, to limit access. The directory's permissions will be set to
+    world writeable/readable/executable since we can't predict what user a
+    container will run as when it accesses the mounted filesystem.
+    """
+    original_umask = umask(0)
+    try:
+        mountroot = mountpoint.parent()
+        if not mountroot.exists():
+            mountroot.makedirs()
+        mountroot.chmod(S_IRWXU)
+        if not mountpoint.exists():
+            mountpoint.makedirs()
+        mountpoint.chmod(S_IRWXU | S_IRWXG | S_IRWXO)
+        mountpoint.restat()
+    finally:
+        umask(original_umask)
```

Comment:
So apparently there is a way to make `umask` not cross-thread, by creating special kind of thread, but that doesn't help us. In any case, though, apparently I don't actually need `umask`? Possibly I thought I did cause of confusion caused by `FilePath` stat caching...